internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,687 @@
+
+func (r PacketCoreControlPlaneResource) ResourceType() string {
+	return "azurerm_mobile_network_packet_core_control_plane"
+}
+
+func (r PacketCoreControlPlaneResource) ModelObject() interface{} {
+	return &PacketCoreControlPlaneModel{}
+}
+
+func (r PacketCoreControlPlaneResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return packetcorecontrolplane.ValidatePacketCoreControlPlaneID
+}
+
+func (r PacketCoreControlPlaneResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"control_plane_access_interface": interfacePropertiesSchema(),
+
+		"site_ids": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: site.ValidateSiteID,
+			},
+		},
+
+		"sku": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(packetcorecontrolplane.BillingSkuGZero),
+				string(packetcorecontrolplane.BillingSkuGOne),
+				string(packetcorecontrolplane.BillingSkuGTwo),
+				string(packetcorecontrolplane.BillingSkuGThree),
+				string(packetcorecontrolplane.BillingSkuGFour),
+				string(packetcorecontrolplane.BillingSkuGFive),
+				string(packetcorecontrolplane.BillingSkuGOneZero),
+			}, false),
+		},
+
+		"user_equipment_mtu_in_bytes": {
+			Type:         pluginsdk.TypeInt,
+			Optional:     true,
+			Default:      1440,
+			ValidateFunc: validation.IntBetween(1280, 1930),
+		},
+
+		"core_network_technology": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(packetcorecontrolplane.CoreNetworkTypeFiveGC),
+				string(packetcorecontrolplane.CoreNetworkTypeEPC),
+			}, false),
+		},
+
+		"platform": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"edge_device_id": {
+						Type:             pluginsdk.TypeString,
+						Optional:         true,
+						DiffSuppressFunc: suppress.CaseDifference,
+						AtLeastOneOf:     []string{"platform.0.edge_device_id", "platform.0.azure_stack_hci_cluster_id", "platform.0.azure_arc_connected_cluster_id", "platform.0.custom_location_id"},
+
+						ValidateFunc: func(i interface{}, s string) ([]string, []error) {
+							// a workaround, sometimes the id from user is with `DataBoxEdgeDevices` instead of `dataBoxEdgeDevices`.
+							// while the service accepts both format.
+							if v, ok := i.(string); ok {
+								i = strings.ReplaceAll(v, "DataBoxEdgeDevices", "dataBoxEdgeDevices")
+							}
+							return devices.ValidateDataBoxEdgeDeviceID(i, s)
+						},
```

Comment:
the context was the user may copy the id from Portal, for this resource this may be a common scenario, so this workaround was requested by svc team. But I agree to remove this as the data source `azurerm_databox_edge_device` has been added.